VirtoCommerce.Storefront/Domain/StaticContent/StaticContentItemBuilder.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using AutoRest.Core.Utilities;
+using Markdig;
+using VirtoCommerce.Storefront.Model.StaticContent;
+
+namespace VirtoCommerce.Storefront.Domain
+{
+    public partial class StaticContentItemBuilder : IStaticContentItemBuilder
+    {
+        internal static readonly Regex headerRegExp = new Regex(@"(?s:^---(.*?)---)");
+        internal static readonly string[] extensions = new[] { ".md", ".liquid", ".html" };
+
+        private readonly IStaticContentItemFactory _factory;
+
+        private readonly List<IContentItemVisitor> _prepareVisitors = new List<IContentItemVisitor>
```

Comment:
I think It would be better to inject all visitors as array of dependencies in the ctor. Why do you need to group them if anyway you call all visitors in the one loop?

Reply:
the visitors order has matter. there is no guarantees to save the order during injecting.
Will it be normal to add priority for visitors?